File: solutions/cpp/protein-translation/1/protein_translation.cpp
#include "protein_translation.h"

namespace protein_translation {
    vector<string> proteins(string input) {
        std::map<string, string> codons_map = {
            {"AUG", "Methionine"},
            {"UUU", "Phenylalanine"}, {"UUC", "Phenylalanine"},
            {"UUA", "Leucine"}, {"UUG", "Leucine"},
            {"UCU", "Serine"}, {"UCC", "Serine"}, {"UCA", "Serine"}, {"UCG", "Serine"},
            {"UAU", "Tyrosine"}, {"UAC", "Tyrosine"},
            {"UGU", "Cysteine"}, {"UGC", "Cysteine"},
            {"UGG", "Tryptophan"},
            {"UAA", "STOP"}, {"UAG", "STOP"}, {"UGA", "STOP"}
        };

        string codon = {};
        string protein = {};
        vector<string> output = {};

        for (unsigned long i = 0; i < (input.length() / 3); ++i) {
            codon = input.substr(3 * i, 3);
            protein = codons_map[codon];
            if (protein == "STOP") break;
            output.push_back(protein);
        }
        return output;
    }
}  // namespace protein_translation

File: solutions/python/protein-translation/1/protein_translation.py
def proteins(strand):
    output = []
    n = 3
    codon_list = [strand[i:i+n] for i in range(0, len(strand), n)]

    protein_codons_dict = {
        'Methionine' : 'AUG',
        'Phenylalanine': ['UUU', 'UUC'],
        'Leucine': ['UUA', 'UUG'],
        'Serine': ['UCU', 'UCC', 'UCA', 'UCG'],
        'Tyrosine': ['UAU', 'UAC'],
        'Cysteine': ['UGU', 'UGC'],
        'Tryptophan': 'UGG',
        'STOP': ['UAA', 'UAG', 'UGA'],
}
    print("codon_list is ", codon_list)
    stop = False
    
    for codon in codon_list:
        if stop == True: break
        for protein in protein_codons_dict:
            if codon in protein_codons_dict[protein]:
                if protein == 'STOP':
                    stop = True
                    break
                else:
                    print('else!')
                    output.append(protein)

    print(output)
    return output